src/utils.py

```diff
@@ -267,13 +267,47 @@ def wrap_atom_predicates(atoms: Collection[LiftedOrGroundAtom],
     return new_atoms
 
 
-def run_policy_until(policy: Callable[[State], Action],
-                     simulator: Callable[[State, Action], State],
-                     init_state: State, termination_function: Callable[[State],
-                                                                       bool],
+def run_policy_until(policy: Callable[[State], Action], env: BaseEnv,
+                     train_or_test: str, task_idx: int,
+                     termination_function: Callable[[State], bool],
                      max_num_steps: int) -> LowLevelTrajectory:
+    """Execute a policy on a train or test task in the environment.
+
+    Note that the environment internal state is updated.
+
+    Terminates when any of these conditions hold:
+    (1) the termination_function returns True,
+    (2) max_num_steps is reached,
+
+    Returns a LowLevelTrajectory object.
+    """
+    state = env.reset(train_or_test, task_idx)
+    states = [state]
+    actions: List[Action] = []
+    if not termination_function(state):
+        for _ in range(max_num_steps):
+            act = policy(state)
+            state = env.step(act)
+            actions.append(act)
+            states.append(state)
+            if termination_function(state):
+                break
+    traj = LowLevelTrajectory(states, actions)
+    return traj
+
+
+def simulate_policy_until(policy: Callable[[State], Action],
+                          simulator: Callable[[State, Action],
+                                              State], init_state: State,
+                          termination_function: Callable[[State], bool],
+                          max_num_steps: int) -> LowLevelTrajectory:
     """Execute a policy from an initial state, using a simulator.
 
+    This is very similar to run_policy_until, with the difference that a
+    simulator (function that takes state as input) is assumed. This function
+    should not be used with any core code, because it violates the environment
```

Comment:
this isn't true -- the environment API does require a simulator to be provided (it's an abstract method).

to remedy this, we could make it not abstract (and update the docstring to explain why). or, we can change this comment to say something else.

Reply:
just change the comment